src/bin/example-metadata.rs

```diff
@@ -0,0 +1,63 @@
+// Licensed to the Apache Software Foundation (ASF) under one
+// or more contributor license agreements.  See the NOTICE file
+// distributed with this work for additional information
+// regarding copyright ownership.  The ASF licenses this file
+// to you under the Apache License, Version 2.0 (the
+// "License"); you may not use this file except in compliance
+// with the License.  You may obtain a copy of the License at
+//
+//   http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+extern crate parquet;
+
+use std::fs::File;
+use std::path::Path;
+
+use parquet::file::metadata::{FileMetaData, ParquetMetaData};
+use parquet::file::reader::{FileReader, SerializedFileReader};
+
+// Run example
+// ===
+// $ cargo run --bin example-metadata
+//
+// Console output
+// ===
+// Number of row groups: 1
+// Version: 1
+// Created by: Some("parquet-mr version 1.8.1 (build 4aba4dae7bb0d4edbcf7923ae1339f28fd3f7fcf)")
+// Number of rows: 4
+// Column INT32: "a"
+// Column INT64: "b"
+// Column BYTE_ARRAY: "c"
+// Column INT32: "d.a"
+// Column INT64: "d.b"
+// Column BYTE_ARRAY: "d.c"
+// Column INT32: "e.list.element"
+//
+fn main() {
```

Comment:
I got:

```
➜  parquet-rs (example) ✔ cargo run --bin example-read
    Finished dev [unoptimized + debuginfo] target(s) in 0.0 secs
     Running `target/debug/example-read`
Reading batch of records: 4
thread 'main' panicked at 'called `Result::unwrap()` on an `Err` value: NYI("Unsupported encoding PLAIN_DICTIONARY")', src/libcore/result.rs:860:4
note: Run with `RUST_BACKTRACE=1` for a backtrace.
```
while running this. Anything I missed?

Reply:
I am not sure. Did you check out my branch? It looks like the problem that I patched the other day. Could you try to fetch latest commits from master?